src/backend/commands/createas.c

```
@@ -590,6 +628,18 @@ static void CreateIvmTriggersOnBaseTables(Query *qry, Node *jtnode, Oid matviewO
 				*relid_map = bms_add_member(*relid_map, rte->relid);
 			}
 		}
+		else if (rte->rtekind == RTE_SUBQUERY)
+		{
+			Query *subquery = rte->subquery;
+			Assert(rte->subquery != NULL);
+			/*
+			 * On the condition, not allow a subquery in the subquery.
+			 */
+			if (in_subquery)
+				elog(ERROR, "neseted subquery is not supported with IVM");
```

in_subquery is only used for this error check. If you can check this in check_ivm_restriction easily, this new argument seems unnecessary. If this is difficult and we have to raise an error here, it would be better to use erport instead of elog.

---

Yes, check_ivm_restriction have already check this. this check doesn't need.
And, CreateIvmTriggersOnBaseTables has more elog() function. so I would fix them,